src/Pages/Checkout/index.js

```diff
@@ -16,20 +28,76 @@ const CartDetail = ({ className, title, price, qty }) => {
 };
 
 const Checkout = () => {
+  const { productState, productDispatch } = useProduct();
+
+  useEffect(() => {
+    wrapAsync(async () => {
+      productDispatch({
+        type: 'ADD_TO_CART',
+        payload: await fetchCartProducts(),
+      });
+    })();
+  }, [productState.cartProducts]);
+
+  const handleIncrement = wrapAsync(async (product) => {
+    await updateProducts(product, 'increment');
+  });
+
+  const handleDecrement = wrapAsync(async (product) => {
+    if (product.qty === 1) {
+      const cartProducts = await removeProductsFromCart(product, 'decrement');
+      productDispatch({
+        type: 'ADD_TO_CART',
+        payload: cartProducts,
+      });
+    } else {
+      await updateProducts(product, 'decrement');
+    }
+  });
+
+  const handleWishlist = wrapAsync(async (product) => {
+    const wishListProducts = await addProductsToWishList(product);
+    productDispatch({
+      type: 'ADD_TO_WISHLIST',
+      payload: wishListProducts,
+    });
+    alert(`${product.title} has been added to wishlist`);
+  });
+
+  const handleDelete = wrapAsync(async (product) => {
+    const cartProducts = await removeProductsFromCart(product);
+    productDispatch({
+      type: 'ADD_TO_CART',
```

Comment:
Yeah, similar to what Nikhil mentioned, here your function is to removeFromCart and dispatch action type is ADD_TO_CART. Maybe you have not tested this yet.